src/ripple/shamap/impl/SHAMap.cpp

```diff
@@ -107,7 +107,7 @@ SHAMap::getStack (uint256 const& id, bool include_nonmatching_leaf) const
     }
 
     if (include_nonmatching_leaf ||
-        (std::static_pointer_cast<SHAMapTreeNode>(node)->peekItem()->getTag() == id))
+        (std::static_pointer_cast<SHAMapTreeNode>(node)->peekItem()->key() == id))
```

Comment:
I get extremely uncomfortable when I see a `dynamic_cast` or `std::static_pointer_cast` without checking for `nullptr`...